pkg/agent/metrics/prometheus.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"k8s.io/klog"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+)
+
+type OVSStatManager struct {
+	ofClient     openflow.Client
+	OVSBridge    string
+	OVSTableDesc *prometheus.Desc
+}
+
+func (c *OVSStatManager) OVSGetStatistics() (
+	ovsFlowsByTable map[string]float64,
+) {
+	ovsFlowsByTable = make(map[string]float64)
+	flowTableStatus := c.ofClient.GetFlowTableStatus()
+	for _, tableStatus := range flowTableStatus {
+		ovsFlowsByTable[strconv.Itoa(int(tableStatus.ID))] = float64(tableStatus.FlowCount)
+	}
+	return
+}
+
+func (c *OVSStatManager) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.OVSTableDesc
+}
+
+func (c *OVSStatManager) Collect(ch chan<- prometheus.Metric) {
+	ovsFlowsByTable := c.OVSGetStatistics()
+	for tableId, tableFlowCount := range ovsFlowsByTable {
+		ch <- prometheus.MustNewConstMetric(
+			c.OVSTableDesc,
+			prometheus.GaugeValue,
+			tableFlowCount,
+			tableId,
+		)
+	}
+}
+
+func NewOVSStatManager(ovsBridge string, ofClient openflow.Client) *OVSStatManager {
+	return &OVSStatManager{
+		ofClient:  ofClient,
+		OVSBridge: ovsBridge,
+		OVSTableDesc: prometheus.NewDesc(
+			"antrea_agent_ovs_flow_table",
+			"OVS flow table flow count.",
+			[]string{"table_id"},
+			prometheus.Labels{"bridge": ovsBridge},
+		),
+	}
+}
+
+func InitializePrometheusMetrics(
```

Comment:
Can this be moved to agent monitor and controller monitor?

Reply:
Monitor packages contain custom CRD implementation. I don't see a reason to mix the above into the same package.
Maybe it's worthwhile to refactor all the monitoring related code into a separate package hierarchy? Is there any value in doing that?